wallet/createtx.go

```diff
@@ -943,22 +935,67 @@ func (w *Wallet) purchaseTickets(op errors.Op, req purchaseTicketRequest) ([]*ch
 		return nil, errors.E(op, errors.Invalid, "stakepool fee percent unset")
 	}
 
+	inSizes := make([]int, 0)
+	outSizes := make([]int, 0)
+	// TODO: use stake.SStxPKHMinOutSize after it has been updated.
+	commitmentScriptSize := 34
+	stakeSubmissionPkScriptSize := 0
+	addrID := [2]byte{req.ticketAddr.ScriptAddress()[0],
+		req.ticketAddr.ScriptAddress()[1]}
+	isP2SHAddr := chaincfg.IsScriptHashAddrID(addrID)
+	isP2PKHAddr := chaincfg.IsPubKeyHashAddrID(addrID)
```

Comment:
These checks are incorrect.  The address ID does not appear in the script, it only appears in the address string representation (addresses have the form `base58(id | payload | checksum)`, while the script will contain only the payload and opcodes to check the payload).  Use type assertions on the ticketAddr field instead.